.github/workflows/preview-pull-request.yml

```diff
@@ -72,28 +49,59 @@ jobs:
+        if: ${{ github.event.workflow_run.event == 'pull_request' && env.PR && env.PR_PREVIEW_REPO && env.DEPLOY_KEY != '' }}
+        id: previous-comment
+        with:
+          issue-number: ${{ env.PR }}
+          comment-author: 'github-actions[bot]'
+          body-includes: pr;head;sha;base
 
       # Comments on PR
-      # TODO: Possible to make this edit, delete, or flag-as-outdated its past comment?
       - name: Comment PR
-        uses: peter-evans/create-or-update-comment@v1
-        if: ${{ github.event.workflow_run.event == 'pull_request' && env.PR }}
+        uses: peter-evans/create-or-update-comment@v2
+        env:
+          DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+        if: ${{ github.event.workflow_run.event == 'pull_request' && env.PR && env.PR_PREVIEW_REPO && env.DEPLOY_KEY != '' }}
         with:
+          comment-id: ${{ steps.previous-comment.outputs.comment-id }}
           issue-number: ${{ env.PR }}
+          edit-mode: replace
           body: |
             Previews, as seen when this [build job](https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}) started (${{ github.event.workflow_run.head_sha }}):
-            [**WebGPU**](${{ steps.deployment.outputs.details_url }}/) <sub>[**`webgpu.idl`**](${{ steps.deployment.outputs.details_url }}/webgpu.idl.txt) | [**Explainer**](${{ steps.deployment.outputs.details_url }}/explainer/) | [**Correspondence Reference**](${{ steps.deployment.outputs.details_url }}/correspondence/)</sub>
-            [**WGSL**](${{ steps.deployment.outputs.details_url }}/wgsl/) <sub>[**`grammar.js`**](${{ steps.deployment.outputs.details_url }}/wgsl/grammar/grammar.js) | [**`wgsl.lalr.txt`**](${{ steps.deployment.outputs.details_url }}/wgsl/wgsl.lalr.txt)</sub>
+            [**WebGPU**](https://${{ github.event.workflow_run.repository.owner.login }}.github.io/${{ env.PR_PREVIEW_REPO }}/gpuweb/gpuweb/${{ env.PR }}/) <sub>[**`webgpu.idl`**](https://${{ github.event.workflow_run.repository.owner.login }}.github.io/${{ env.PR_PREVIEW_REPO }}/gpuweb/gpuweb/${{ env.PR }}/webgpu.idl.txt) | [**Explainer**](https://${{ github.event.workflow_run.repository.owner.login }}.github.io/${{ env.PR_PREVIEW_REPO }}/gpuweb/gpuweb/${{ env.PR }}/explainer/) | [**Correspondence Reference**](https://${{ github.event.workflow_run.repository.owner.login }}.github.io/${{ env.PR_PREVIEW_REPO }}/gpuweb/gpuweb/${{ env.PR }}/correspondence/)</sub>
+            [**WGSL**](https://${{ github.event.workflow_run.repository.owner.login }}.github.io/${{ env.PR_PREVIEW_REPO }}/gpuweb/gpuweb/${{ env.PR }}/wgsl/) <sub>[**`grammar.js`**](https://${{ github.event.workflow_run.repository.owner.login }}.github.io/${{ env.PR_PREVIEW_REPO }}/gpuweb/gpuweb/${{ env.PR }}/wgsl/grammar/grammar.js) | [**`wgsl.lalr.txt`**](https://${{ github.event.workflow_run.repository.owner.login }}.github.io/${{ env.PR_PREVIEW_REPO }}/gpuweb/gpuweb/${{ env.PR }}/wgsl/wgsl.lalr.txt)</sub>
```

Comment:
nit: `https://${{ github.event.workflow_run.repository.owner.login }}.github.io/${{ env.PR_PREVIEW_REPO }}/gpuweb/gpuweb/${{ env.PR }}` is reused a lot, possible to precompute that in an earlier step so it cas just be something like `${{ env.PREVIEW_BASE_URL }}` instead?

Reply:
@kainino0x Noted for the revision PR!